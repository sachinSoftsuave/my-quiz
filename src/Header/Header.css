.header{
    display:flex;
    flex-wrap: wrap;
    flex-direction: column;
    justify-content: flex-start;
    justify-content: space-between;
    width: 100%;
    background-color: dodgerblue;
    height: 80px;
    font-weight:900px;
    font-family: Verdana, Geneva, Tahoma, sans-serif;
    font-size: 15px;
    color:white;
}

.title1{
    padding-left: 20px;
    margin-left:20px;
    padding-top:30px;
    cursor:pointer;
    color:white;
    text-decoration: none;
}
.title2{
    padding-left: 20px;
    margin-left:1250px;
    padding-top:15px;
}



