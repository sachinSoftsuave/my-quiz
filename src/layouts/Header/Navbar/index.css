.linkHighScore{
     cursor:pointer;
     text-decoration: none;
     font-weight:900px;
     font-family: Verdana, Geneva, Tahoma, sans-serif;
     font-size: 15px;
     color:white;
 }