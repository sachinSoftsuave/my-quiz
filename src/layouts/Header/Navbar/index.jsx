import React from "react";
import './index.css'

function Navbar() {
  return (
    <div>
      <a className="linkHighScore" href="./highScore">
        View Highscore <i className="fas fa-hand-point-left fa-lg"></i>
      </a>
    </div>
  );
}

export default Navbar;
