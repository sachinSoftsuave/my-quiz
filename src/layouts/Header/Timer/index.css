.timer{
     cursor: pointer;
     text-decoration: none;
     font-weight: 900;
     font-family: Verdana, Geneva, Tahoma, sans-serif;
     font-size: 15px;
     color: white;
 }