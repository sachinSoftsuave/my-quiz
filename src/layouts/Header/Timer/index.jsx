import React, {useState, useEffect, useContext} from "react";
import { useNavigate } from "react-router-dom";
import { QuizContext } from "../../../hooks/context";
import "./index.css";

function Timer() {

  const [minutes, setMinutes] = useState(50);
  const [isActive, setIsActive] = useState(false);
  const {isQuizStarted} = useContext(QuizContext);
  let navigate = useNavigate();
  console.log(isQuizStarted,"isQuizStarted");

  useEffect(() => {
    if(isQuizStarted)
    {
      setMinutes(50)
      startTimer()
    }
  },[isQuizStarted])

  function startTimer() {
    setIsActive(!isActive);
  }

  useEffect(() => {
    let interval = null;
    if (isActive && minutes > 0) {
      interval = setInterval(() => {
        setMinutes(minutes => minutes - 1);
      }, 1000);
    } else if(minutes <=0) {
      clearInterval(interval);
      setIsActive(false);
      navigate('/results')
    }
    return () => clearInterval(interval);
  }, [isActive, minutes]);

  return (
    <div>
      <p className="timer"> Time:{minutes} </p>
    </div>
  );
}

export default Timer;
