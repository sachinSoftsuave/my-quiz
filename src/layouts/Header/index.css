.header{
     display:flex;
     flex-wrap: wrap;
     flex-direction: row;
     justify-content: space-between;
     background-color: #358380;
     align-items: center;
     padding: 20px 30px;  
}