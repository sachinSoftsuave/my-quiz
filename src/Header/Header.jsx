import React from 'react';
import './Header.css';
 
 function Header(){

    return(
        <div className='header'>
            <a className='title1' href='./HighScore'>View Highscore <i className="fas fa-hand-point-left fa-lg"></i></a>
            <p className='title2'> Time: </p>
        </div>
    )
}

export default Header;