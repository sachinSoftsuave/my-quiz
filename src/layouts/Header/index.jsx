import React from "react";

import './index.css'
import Navbar from "./Navbar";
import Timer from "./Timer";

function Header() {

  return <div className="header">
       <Navbar/>
       <Timer/>
  </div>;
}

export default Header;
