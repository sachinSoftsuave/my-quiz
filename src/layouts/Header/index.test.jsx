import React from "react";
import { shallow } from 'enzyme';
import "jsdom-global/register";
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Header from "./index.jsx";
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useContext: () => ({
  isQuizStarted: false
  }),
}));

describe("header component", () => {
  
  it('it render header component', () =>{
        const wrapper  = shallow(<Header/>);
        expect(wrapper.exists()).toBe(true);
  })

   it('it renders the snapshot of header component', () =>{
        const wrapper  = shallow(<Header/>);
       expect(toJson(wrapper)).toMatchSnapshot();
  })

});
