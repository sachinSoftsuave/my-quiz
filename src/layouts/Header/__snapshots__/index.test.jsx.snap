// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component it renders the snapshot of header component 1`] = `
<div
  className="header"
>
  <Navbar />
  <Timer />
</div>
`;
